pkg/collector/corechecks/sbom/processor.go

```diff
@@ -10,15 +10,17 @@ import (
 	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/epforwarder"
 	"github.com/DataDog/datadog-agent/pkg/tagger"
 	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
 	queue "github.com/DataDog/datadog-agent/pkg/util/aggregatingqueue"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
-	"google.golang.org/protobuf/types/known/timestamppb"
 
-	"github.com/DataDog/agent-payload/v5/sbom"
 	model "github.com/DataDog/agent-payload/v5/sbom"
+
+	"google.golang.org/protobuf/proto"
```

Comment:
This PR uses both google protobuf and the deprecated gogo. Would it make sense to use one library everywhere?

Reply:
According to my experiments, attempting to (un)marshal a structure with a different library than the one that was used to generate the protobuf structs leads to a panic (see #15172 for an example).
Whereas [`contimage`](https://github.com/DataDog/agent-payload/blob/d60f4b03a1f62eaea4981adbdebebe199d1cb5b0/Rakefile#L100) and [`sbom`](https://github.com/DataDog/agent-payload/blob/d60f4b03a1f62eaea4981adbdebebe199d1cb5b0/Rakefile#L104) messages are generated by the Google protobuf, [`contlcycle`](https://github.com/DataDog/agent-payload/blob/d60f4b03a1f62eaea4981adbdebebe199d1cb5b0/Rakefile#L94) is still generated by the deprecated gogo protobuf.

In the past, I already tried to migrate all the `agent-payload` messages from the deprecated gogo protobuf to google’s one (see DataDog/agent-payload#213 and DataDog/datadog-agent#14844), but it was much more painful than I originally thought.
The main reason is that, compared to the gogo generated structs, the google generated ones are having private hidden fields that are containing a `sync.Mutex` field.

The new hidden private fields are making all the structs comparisons done in tests fail.
It also makes their use as keys of map hazardous.
The `sync.Mutex` field makes the generated structs not copiable. So, it breaks each time such a struct is passed as a function argument or as a return value of a function without a pointer.

What I can do is to work on migrating `contlcycle` message from the deprecated gogo protobuf to the google one in a separate dedicated PR.

Reply:
I’ve opened DataDog/agent-payload#239 to migrate `contlcycle` to the official protoc go generator.
I’ll do the adaptation in `datadog-agent` once this PR is merged.

Reply:
And here is the follow up PR to migrate from gogo to the official google protobuf for `contlcycle` message: #16168.